### ISSUE TITLE
Garbled characters occur when extracting text from Chinese PDF documents

### ISSUE THREAD

**Issue by Deng-yongbiao** [podofo/podofo#216]
Fill the form below, removing parts that don't apply, including this note.
Please note that PoDoFo is currently activelly developed and maintained only
by a single person. The quickest and safest way to ensure bugs that
affects you are fixed and the features you need are checked-in is to
[contribute](https://github.com/podofo/podofo#contributions) high-quality patches.

* **I'm submitting a:**
  - [ ] Vulnerability report -> **use the specific [form](https://github.com/podofo/podofo/security/advisories/new) instead!**
  - [x] Bug report
  - [ ] Feature request/suggestion

* **[Bug report] What is the current behavior?**
Hey! I am trying to use PODOFO to extract text content from a pdf, The code I use is as follows
  ```cpp
  TEST_CASE("MyTextExtraction"){
      PdfMemDocument doc;
      doc.Load(TestUtils::GetTestInputFilePath("/Users/shimu/Desktop/GBK-EUC-H.pdf"));
      auto& page = doc.GetPages().GetPageAt(0);
      vector<PdfTextEntry> entries;
      page.ExtractTextTo(entries);
      for (const auto& entry : entries){
          cout << entry.Text << endl;
      }
  }
  ```

But the program output is some garbled code like:
  ```text
  2007Äê ÈÕ±¾ °ä ²¼ ¡¶º£Ñó »ù ±¾ ·¨ ¡·Ðû²¼ ÊµÊ© ¡°ÐÂ º£ Ñó Á¢ ¹ú¡±
  Õ½ ÂÔ £¬ ÈÕ±¾ Ö÷ Òª ´Ó Èý ·½ Ãæ ÍÆ ½ø Õâ Ïî Õ½ ÂÔ £º Ò» ÊÇ ¹Ø ÓÚ º£ Ñó ×Ê Ô´ ÓÈ
  Æä ÊÇ º£µ× ×ÊÔ´ ¼°Æä ¿ª A-E£®Êõ µÄµ÷ ²éÑÐ ·¢ £»¶þÊÇ Îª È¦ Õ¼×¨ Êô ¾\x00¼Ã
  Çø¼° ´ó Â½ ¼Ü £¬¶Ô»ù µã º£ µº½ø ÐÐµ÷ ²é ¡¢²â Á¿ ¡¢ÖÆ Í¼¼° ¡°±£ È« ¡± £»
  Èý ÊÇ Ôö ¼Ó ¡°È¦ º£ ¡± µÄ Êµ Á¦ ¡¢ Ìå ÖÆ ½¨ Éè ºÍ ¾ü Á¦ ²¼ ¾Ö £¬Ôö ¼Ó º£ ÉÏ ×Ô
  ÎÀ ¶Ó ¼° º£ ÉÏ ±£ °² Ìü µÄ ·É »ú ¡¢ ½¢ Í§ µÈ Êµ Á¦ Åä ÖÃ ÊÇ ÖØ ÖÐ Ö® ÖØ ¡£ 2010
  Äê ÈÕ±¾ º£ÉÏ ±£ °²Ìü ÔÚ µö Óã µºº£Óò ×¥ ¿Û ÖÐ ¹úÓæ´¬ µÄ ÊÂ¼þ £¬¾ÍÊÇ Æó
  Í¼ ÖÆ Ôì Âä Êµ º£ Ñó ·¨ ÖÆ µÄ ¡°Ö´ ·¨ ¡± °¸ Àý ¡£ ÈÕ±¾ ¶Ô µö Óã µº ¼° Æä ¸½ Êô
  µº Óì ²É È¡ ¡°Ãü Ãû ¡± ¡¢ ¡°ÎÞ Ö÷ ÍÁ µØ ¡± ÊÕ ¹é ¡°¹úÓÐ ¡± µÈ ¼¤ »¯ Ã¬ ¶Ü
  µÄ´ë Ê© £¬ÕÃ ÏÔ Æä Ïòº£ Ñó½ø ÐÐ ¾üÊÂ ºÍ ¾\x00 ¼ÃÀ© ÕÅ µÄÕ½ ÂÔ Òâ Í¼¡£ Á½Äê
  À´ £¬ ÈÕ±¾ Í¶ Èë ¾Þ ×Ê ¼Ó Ç¿ º£ ÉÏ ×ÔÎÀ ¶Ó ºÍ º£ ÉÏ ±£ °² Ìü µÄ ¡°Ó² ¼þ ¡± ½¨
  Éè £¬¼Ó ËÙ ÍÆ £ºL&4oo Ïî º£ Ñó Õ½ ÂÔ Ïî Ä¿ £¬¸² ¸Ç ·¶ Î§ ¿í ·º £¬ÍÆ ½ø ´Î Ðò
  Ã÷ È· £¬Í¶ Èë ²Æ Á¦ ¾Þ ´ó £¬ Êµ Ê© Á¦ ¶È Ñ¸ ÃÍ ¡£
  Îª ÁËÂä Êµ ¡¶º£ Ñó»ù ±¾ ·¨ ¡·È· ±£ º£ Ñó°²È« µÄ¸÷ Ïî¹æ ¶¨ ²¢µ÷ Õû
  ·À ÎÀ Õ½ ÂÔ £¬201OÄê 12ÔÂ £¬ ÈÕ±¾ ÒÔ ÖÐ ¹ú Îª ¡°¼Ù Ïë µÐ ¡± ÐÞ ¶© ÁËÐÂ ·À
  ÎÀ ¼Æ »® ´ó ¸Ù ¡£ ÐÂ ´ó ¸ÙÌá ³ö £º¾¡ ¹Ü ·¢ Éú ´ó ¹æ Ä£ Õ½ Õù µÄ ¿É ÄÜ ÐÔ ½µ
  µÍ £¬µ« Î» ÓÚ ¡° ¡®Æ½ Ê± ¡¯Óë ¡®ÓÐÊÂ ¡¯Ö® ¼äµÄ ÖÐ¹úº£ ¾ü¼° º£ ¼àµÈ µÄ
  º£ Ñó »î ¶¯ Óë ÇÖ º¦ ÈÕ±¾ È¨ Òæ ÓÐ ¹Ø ¡± £¬ ±Ø Ðë Êµ Ê© ¡°¶¯ Ì¬¶ô ÖÆ ¡± £»Òª
  ÊµÏÖ ÃñÖ÷ µ³³« µ¼ µÄ ¡°ÈÕÃÀ¶Ô µÈ ¡± £¬Éî»¯ ÈÕÃÀ Í¬ÃË ºÏ×÷ £¬Òª Çó×¤
  ÈÕ ÃÀ ¾ü ½« Ö§ Ô® ÈÕ±¾ ×÷ Îª ¡°Ê× Òª ¡± ÈÎ Îñ Ò» »» ÑÔ Ö® £¬Òª Çó ÃÀ ¾ü °Ñ
  Ð\x00 ·Àµö Óã µº·Å ÔÚ Ê×Î» ¡£°´ ÕÕ ÉÏÊö Õ½ ÂÔ°² ÅÅ £¬ ÈÕ±¾ ²»µ« ÔÚ ÖÐ ÈÕÁì
  ÍÁ Õù Òé ÖÐÓÐ ÊÑ ÎÞ ¿Ö £¬ ¶ø ÇÒ È« Á¦ Åä ºÏ ÃÀ
  ¹úÔÚ ÄÏ º£ ½Á ¾Ö £º¹Ä Àø Ö§ ³Ö ·Æ ÂÉ ±ö ¡¢ Ô½
  ÄÏ µÈ Éù Ë÷ ¹úÔÚ ¶«ÃË ÏµÁÐ »á Òé ÉÏ ¡°°ó
  ¼Ü ¶«ÃË ¡± Î§¹¥ ÖÐ ¹ú¡£°é Ëæ×Å ÃÀ ¹úÑÇ Ì«
  ¡°ÔÙ Æ½ ºâ ¡± Õ½ ÂÔ µÄ ½ø Õ¹ £¬ ÈÕ±¾ Í¹ ÏÔ Óë
  ÃÀ ¹ú ¡°ºÏ »ï ¡± Ö÷µ¼ ÑÇ Ì«°² È«ÊÂ ÎñÈ· Á¢
  µØ Çø ×¼ °Ô È¨ µÄ Õ½ ÂÔ Òâ Í¼ ¡£
  ´Ó ÈÕ±¾ ¹ú ÄÚ Çé ¿ö À´ ¿´ £¬ ºÁ ÎÞ Ö´ Õþ
  ¾\x00 Ñé µÄ Ãñ Ö÷ µ³ ÉÏ Ì¨ Ö´ Õþ Óë ÌØ ´ó Ìì ÔÖ ÖØ
  ºÏ £¬ÊÇ µ¼ÖÂ ÈÕ±¾µ÷ ÕûÕ½ ÂÔ £¬ÖÐ ÈÕ¹Ø Ïµ
  ¶ñ »¯ µÄ ÖØ Òª Ô\x00 Òò ¡£ 2011Äê ¡°3 ¡¤11¡±
  ÌØ ´ó µØ Õð ¡¢ º£ Ð¥ ¼° ºË Ð¹ Â© Èý ÖØ ÔÖ º¦ µ¼
  ÖÂ ÈÕ±¾ ¹úÃñÐÄ ÀíÒì ³£´à Èõ £¬ÝÑÖ±ÈË Õþ
  ¸® Ó¦ ¶Ô ²» Àû Óú ·¢ ¼Ó ¾ç ÁËÕþ ¾Ö »ì ÂÒ ¡£ ÃñÖ÷ µ³Õþ ¸® Óë µØ ·½ ×Ô ÖÎÕþ ¸®
  Ö® ¼ä µÄ ¡°ÈË Âö ¹Ø Ïµ¡± Ô\x00 ±¾ ½ü ºõ ¿Õ °× £¬ Ò° Ìï ÄÚ ¸ó Îª Êµ ÏÖ Ôö Ë° µ¼ ÖÂ
  Ãñ Ö÷ µ³ ÄÚ ²¿ ·Ö ÁÑ £¬ ¸ü ¼Ó ÎÞ Á¦ ½Ú ÖÆ µØ ·½ ¡£ 2012#-ÒÔ À´ £¬Ê¯ Ô\x00 É÷ Ì«
  ÀÉ µÈµØ ·½ÐÐ Õþ ³¤ ¹Ù ÎÞÊÓ ÖÐ¹úÕþ ¸® ÒÔ ¼ÍÄî ÖÐ ÈÕ°î ½» Õý³£ »¯40ÖÜ Äê
  Îª Æõ »ú ÍÆ ¶¯ ÖÐ ÈÕ Õ½ ÂÔ »¥ »Ý ¹Ø Ïµ µÄ Á¼ ºÃ Òâ Ô¸ £¬½Ó Á¬ ÔÚ ·ñ ÈÏ ¡°ÄÏ ¾©
  ´óÍÀ É± ¡± ¡¢ ¡°¹º Âò¡±µö Óã µºÁ½ ´óÕþ ÖÎÃô ¸Ð ÎÊÌâ ÉÏ ½ø ÐÐÌô ÐÆ £¬ÀÌ
  È¡Õþ ÖÎ ×Ê ±¾ £¬µ¹ ±ÆÒ° ÌïÕþ ¸® ¡£ ÔÚ¾è ¿î Ñ¸ËÙ Ôö ¼Ó £¬ Ãñµ÷ ºÍÓß ÂÛ Öú
  ÍÆ £¬ÔÚ Ò° µ³ ÅÔÇÃ²à »÷µÄ ´Ì¼¤ ÏÂ £¬½è Ê¯ Ô\x00 ¡°Âòµº ¡± Æõ»ú³¹ µ× ½â ¾ö
  µö Óã µº¼° Æä ¸½Êô µºÓì µÄ ¡°¹úÓÐ »¯¡± ÎÊÌâ ÒÑÔÚÕþ ½ç ÐÎ ³É ¹²Ê¶ ¡£ ÄÚ
  ¸óÐÐ ½« ¾ÍÄ¾ Ö® ¼Ê£¬ ÒàÊÇÕþ ¿Í Æ´ ¡°ËÀ ¡± Ò»²© Ö® Ê± £¬ ÒòÎªËû ÃÇ ÏÂ Ì¨
  ºó ²»»á ¸ºÔðÊÕÊ° ×Ô¼ºÁô ÏÂµÄ ÀÃÌ¯ ×Ó ¡£¹Ã ÇÒ ²»ÂÛ °Ñ ³Ö È¨±ú ÕßµÄ Ò»
  ¼ºË½ Àû £¬ º£ Ñó Õ½ ÂÔ ÊÇ ÈÕ±¾ 21ÊÀ ¼Í ×î ÖØ Òª µÄ ¹ú ¼Ò ×Û ºÏ ·¢ Õ¹ Õ½ ÂÔ £¬
  ¼ÄÍÐ ×Å ÈÕ±¾ Ïòº£ ÑóÒª ¡°¹úÍÁ ¡± ¡¢Òª ×Ê Ô´ ¡¢Òª ¾Í Òµ ¡¢Òª ¾\x00 ¼ÃÔö ³¤
  µÄºñ Íû ¡£ÔÚ ÓÒ Òí ÊÆ Á¦µÄ ÑÛ ÖÐ £¬Îª »ñÈ¡ ÈÕ±¾ ¹úÍÁ Ãæ»ý 12±¶ µÄ ¡°º£
  Ñó ¹ú ÍÁ ¡± £¬ ¼´ Ê¹ ÆÆ »µ ÖÐ ÈÕÕ½ ÂÔ »¥ »Ý ¹Ø Ïµ Éõ ÖÁ ÉóIÆð Îä Á¦ Õù ¶Ë £¬ Ò²
  ²» ¹ý ÊÇ ¡°¶ª Ö¥ Âé ±£ Î÷¹Ï ¡± ¡£ Òò Îª Ëû ÃÇ ×ÔÖª ÈÕ±¾ µÄ ÈË ¿Ú ¡¢ ²Æ Á¦ ¼°
  ¹ú¼ÊÓ° Ïì ÕýÔÚ Óë Ê±¾ã ¼õ £¬ ¡°Ê± ¼ä¡± ²» ÔÚ ÈÕ±¾ Ò» ±ß ¡£ÔÚ ÈÕ±¾º£ Ñó
  Õ½ ÂÔ Óë ·À ÎÀ Õ½ ÂÔ ÖØ ºÏ £¬ ¹ú ¼Ò Õ½ ÂÔ °² ÅÅ Óë µØ ·½ Õþ ¸® µ¹ ±Æ ÖØ ºÏ ¡¢ Ìì
  ÔÖ Óë ¡°ÈË »ö ¡± ÖØ Éá µÄ ½»Ïà ×÷ ÓÃ ÏÂ £¬ ÖÐ ÈÕ ¹Ø Ïµ Ãæ ÁÙ ¿Õ Ç° ÑÏ ¾þ µÄ ¾Ö
  Ãæ £¬Õý ÔÚ ÔÐ ÓýÇ° Ëù Î´ÓÐ µÄÎ£ »ú £¬ÎÒ ¹ú²» ÄÜ²»×ö ºÃ ¸÷ ·½ Ãæ µÄÓ¦ ¶Ô
  ×¼ ±¸
  24 1ÈËÃñÂÛÌ³ I2012£®10ÉÏ
  ```
  In addition, a warning was printed as follows:
 ```text
WARNING: Unable to find font object F0
WARNING: Unable to provide a space size, setting default font size
WARNING: Unable to find font object F0
WARNING: Unable to provide a space size, setting default font size
WARNING: Unable to find font object F0
...
 ```

* **[Bug report] What is the expected behavior?**
  I now know that the text in this PDF is in Chinese. What I want is to be able to correctly extract the UTF-8 encoding corresponding to Chinese, but the effect is obviously not achieved so far.

I have tested hundreds of Chinese PDF documents, and there are probably a few PDF documents that have this garbled problem.
* **[Bug report] Please provide the steps to reproduce and if possible a minimal reproduction code of the problem**

* **[Feature request] What feature should be implemented in PoDoFo?**
I hope to be able to solve some garbled text extraction problems in Chinese PDF documents. Or provide some solutions
* **Please tell us about your environment:**
  - Version/git revision: [0.10.4]
  - Operating System: all
  - Package manager used: [source]

* **[Bug report/Feature request] Other information**
this is the sample
[GBK-EUC-H.pdf](https://github.com/user-attachments/files/17931070/GBK-EUC-H.pdf)

  E.g. sample PDF documents, stacktraces, related issues, links for us to have contex, suggestions how to fix, etc.

**Comment by ceztko** [comment]
Please, try with master. A recent [issue](https://github.com/podofo/podofo/issues/144) has been closed recently and should have the features needed to extract correctly that text. Please note that there will not be a backport to 0.10.x (it will be released together 1.0.0 soon).

**Comment by ceztko** [comment]
I tried myself and I can decode Chinese text on that document with master.
